native/game_backend/src/player.rs

```diff
@@ -77,19 +80,36 @@ impl Player {
         self.kill_count += 1;
     }
 
-    pub fn move_position(&mut self, config: &Config) {
+    pub fn move_position(&mut self, config: &Config, grid: &SpatialHashGrid) {
         // A speed of 0 (or less) means the player can't move (e.g. paralyzed, frozen, etc)
         if self.speed == 0 || !self.moving {
             return;
         }
 
         self.add_action(Action::Moving, 0);
-        self.position = map::next_position(
+        let possible_next_position = map::next_position(
             &self.position,
             self.direction,
             self.speed as f32,
             config.game.width as f32,
         );
+
+        let ids_in_grid_for_player = grid.get_nearby(&GameEntity::from(&(self.clone())));
```

Comment:
Since `GameEntity::from` receives a reference, let's just do:
```suggestion
        let ids_in_grid_for_player = grid.get_nearby(&*self.clone());
```
Cloning `Player` each time may be too costly.

Reply:
(Rationale: there's at least 6 calls to `malloc` per `Player::clone` call)